app/services/submit_application.rb

```diff
@@ -24,7 +24,7 @@ def call
 private
 
   def send_reference_request_email_to_referees(application_form)
-    application_form.application_references.includes(:application_form).each do |reference|
+    application_form.application_references.where(feedback: nil).each do |reference|
```

Comment:
The presence of feedback doesn't mean the reference was submitted (the submission is a separate step now). This might be more robust:

```suggestion
    application_form.application_references.not_requested_yet.each do |reference|
```